control_record.go

```diff
@@ -0,0 +1,70 @@
+package sarama
+
+type ControlRecordType int
+
+const (
+	ControlRecordAbort ControlRecordType = iota
+	ControlRecordCommit
+	ControlRecordUnknown
+)
+
+// Control records are returned as a record by fetchRequest
+// However unlike "normal" records, they mean nothing application wise.
+// They only serve internal logic for supporting transactions.
+type ControlRecord struct {
+	Version          int16
+	CoordinatorEpoch int32
+	Type             ControlRecordType
+}
+
+func (cr *ControlRecord) decode(key, value packetDecoder) (err error) {
```

Comment:
I have been trying to stay away from naked returns, they make code confusing and hide bugs sometimes, if you can remove the naked return that would be great.